Add releasing packages documentation (#1689)
AcademySoftwareFoundation/rez: docs/source/releasing_packages.rst

==================
Releasing packages
==================

Rez packages can be built and deployed to the :data:`release_packages_path`
using the :ref:`rez-release` tool. This tool performs the following actions:

* All :doc:`actions <building_packages>` that the `rez-build` tool performs.
* Executes all configured :data:`release_hooks`.
* Executes all configured `release_vcs`.

> This tool generally runs the same as :ref:`rez-build` otherwise. For more

Reviewer: Yes and no. It's true that it's somewhat similar, but there is still differences, for example, see https://rez.readthedocs.io/en/stable/package_definition.html#tests. There might also be other differences. For example, a release might result in stricter file permissions (though I don't remember clearly).

Reviewer: Basically, I would like to make it clear that that is advantages to use `rez-release`. Though, I have to say that I can't articulate what the advantages are. In other words, when someone reads this page, what will be their conclusion? Will it be that they don't need `rez-release` since it's similar to rez-build` or will it be confusion as to when to use one vs the other?

Author: @JeanChristopheMorinPerso Can you clarify what you differences you're referring to with the `tests` section of the `package_definition` page?

Reviewer: You can set `run_on` on a test to `pre_release` for example. This test would only run before a release.

==================
Releasing packages
==================

Rez packages can be built and deployed to the :data:`release_packages_path`
using the :ref:`rez-release` tool. This tool performs the following actions:

* All :doc:`actions <building_packages>` that the ``rez-build`` tool performs.
* Executes all configured :data:`release_hooks`.

> * Executes all configured `release_vcs`.

Reviewer: ```suggestion
* Executes all configured :gh-rez:`src/rezplugins/release_vcs` plugins.
```
?

When to release vs build
============================
Packages *can* be installed to the :data:`release_packages_path` manually

> by running ``rez-build <release_packages_path>``, so why would you use

Reviewer: ```suggestion
by running :option:`rez-build -i --prefix /path/to/repo <rez-build --prefix>`, so why would you use
```

``rez-release``?

Well, here are a few benefits of ``rez-release``:

* The package will automatically go to the configured :data:`release_packages_path`, whereas :ref:`rez-build` will go to the :data:`local_packages_path` by default.
* Tests being run by with the :ref:`rez-test` tool can run specifically prior to release, ensuring that releases pass any configured tests first.
* Automatic sanity checks to ensure local repo is ready for release.
* Automatic VCS tagging.

> * Many :ref:`helpful package attributes <release-package-attributes>` added to the released `package.py` at build time.

Reviewer: ```suggestion
* Many :ref:`helpful package attributes <release-package-attributes>` are automatically added to the released package definition file.
```
